browse.py

```diff
@@ -315,3 +317,46 @@ def _filter_vault_deposit_index(row):
     # Filter out deposit vault index collection
     return not re.match("^/[^/]+/home/vault-[^/]+/deposit-[^/]+/index$",
                         x['COLL_NAME'])
+
+
+@api.make()
+def api_load_text_obj(ctx, file_path='/'):
+    """Retrieve a text file (as a string) in either the research, deposit, or vault space.
+
+    :param ctx:       Combined type of a callback and rei struct
+    :param file_path: Full file path of file to load
+
+    :returns: file as a string or API status in case of error
+    """
+    # Obtain some context.
+    # - What kind of collection path is this?
+    space, _, _, _ = pathutil.info(file_path)
+    if space not in [pathutil.Space.RESEARCH, pathutil.Space.DEPOSIT, pathutil.Space.VAULT]:
+        return api.Error('invalid_space', 'The given space is not a valid space (should be research, deposit, or vault)')
+
+    # Check exists
+    if not data_object.exists(ctx, file_path):
+        return api.Error('nonexistent', 'The given path does not exist')
+
+    # Check end extension
+    valid_extension = False
+    for ext in config.text_file_extensions:
+        ending = '.' + ext
+        if file_path.lower().endswith(ending):
+            valid_extension = True
+
+    if not valid_extension:
+        return api.Error('not_valid', 'The given data object does not have a valid file extension')
```

Comment:
I'm not sure how to pick names for the api errors.

Reply:
I think you did great :)